Library/src/Multiplayer/SpaceEntitySystem.cpp

```diff
@@ -378,6 +378,18 @@ void SpaceEntitySystem::DestroyEntity(SpaceEntity* Entity, CallbackHandler Callb
 														  },
 														  Components};
 
+
+	auto EntityComponents = Entity->GetComponents();
+	auto Keys			  = EntityComponents->Keys();
+
+	for (size_t i = 0; i < Keys->Size(); ++i)
+	{
+		auto EntityComponent = Entity->GetComponent((*Keys)[i]);
+		EntityComponent->OnLocalDelete();
```

Comment:
Since we use the verb destroy when it comes to removing entities from spaces, should we use the same verb for components? ie, should this function be called `OnLocalDestroy`?

Reply:
I'll update

Reply:
Actually, I decided to use delete here to indicate that this should be called when the compoennt is actually deleted from the space. We "Destroy" components even when tearing down CSP when exiting a space.

Reply:
Ok so
Destroy - removes component _locally_
Delete - removes component _locally and remotely_

Makes sense. Please do find a place to document these terms in GH.